dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -230,10 +230,7 @@ public void AddSteps(params ISKFunction[] steps)
     /// </remarks>
     public Task<Plan> RunNextStepAsync(IKernel kernel, ContextVariables variables, CancellationToken cancellationToken = default)
     {
-        var context = new SKContext(
-            kernel,
-            variables,
-            kernel.Functions);
+        var context = kernel.CreateNewContext(variables);
```

Comment:
Looks like we create this context just so we can pass the `kernel` and `variables` to `InvokeNextStepAsync`

Inside `InvokeNextStepAsync` it looks like we only use this new `context` to access the values, so why not pass `kernel` and `variables` directly?

```

Reply:
This will be changed in a future PR update. (Phase 3 and 4 will be changing Plans and Template engines).

Reply:
@lemillermicrosoft FYI